code/modules/power/smes.dm

```diff
@@ -412,6 +423,21 @@
 	if(!output_attempt)
 		outputting = 0
 
+/obj/machinery/power/smes/take_damage(var/amount)
+	amount = max(0, round(amount))
+	damage += amount
+	if(damage > maxdamage)
+		visible_message("<span class='danger'>\The [src] explodes in large rain of sparks and smoke!</span>")

```

Comment:
`in a large shower of`, I think. It's definitely missing an article, and although I haven't personally ever heard "rain of sparks" used over "shower" they're synonyms, so I don't think it particularly matters which one is used.